src/iface/ethernet.rs

```diff
@@ -231,6 +232,13 @@ impl<'a, 'b, 'c, DeviceT> Interface<'a, 'b, 'c, DeviceT>
                             device_result = inner.dispatch(tx_token, timestamp, Packet::Tcp(response));
                             device_result
                         }),
+                    #[cfg(feature = "proto-icmpv4")]
+                    Socket::Icmpv4(ref mut socket) =>
+                        socket.dispatch(&caps, |response| {
+                            let tx_token = device.transmit().ok_or(Error::Exhausted)?;
+                            device_result = inner.dispatch(tx_token, timestamp, Packet::Icmpv4(response));
+                            device_result
+                        }),
```

Comment:
This match statement *begs* being put in a macro.

Reply:
+1

Reply:
Only comment I haven't fixed yet. Thinking about how to best accomplish this since some branches need `ChecksumCaps` etc.